### PR TITLE
refactor: eslint fixes

### PR DESCRIPTION
**Pull request by pirhoo**
This PR aims at applying eslint suggestions before adding eslint/prettier to the CI (in another PR).

### CODE REVIEW

File: lib/components/DigitsInput.vue [hunk 1]
```diff
@@ -107,6 +107,7 @@
       joinedValues (): string {
         return filter(this.values, v => !isNaN(v as any)).join('')
       },
+      /* eslint-disable no-undef */
```
Comment (marieglr) [inline lib/components/DigitsInput.vue:110]: why is this necessary?

**Review by marieglr — APPROVED** [review]
:nail_care:

File: package.json [hunk 2]
```diff
@@ -110,7 +110,9 @@
       "eslint:recommended",
```
Comment (caro3801) [inline package.json:110]: Since this PR is about linting I would enforce it using "plugin:vue/recommended" instead of plugin:vue/essential as in  
https://github.com/ICIJ/datashare-client/blob/master/.eslintrc.js

Comment (caro3801) [inline package.json:110]: The order of the plugins goes from the most generic (eslint) to the most specific (typescript ) for rules to be override
```suggestion
 "extends": [
      "eslint:recommended",
      "plugin:vue/essential",
      "@vue/typescript"
    ],
    ```
  
https://eslint.vuejs.org/user-guide/#configuration

File: lib/components/SharingOptions.vue [hunk 3]
```diff
@@ -1,198 +1,259 @@
...
-        }
-        return get(document.head.querySelector(metaSelector), 'content', config.get(key))
-      }
+    /**
+     * Sharing contents which can be generic (<em>title</em>, <em>description</em>, etc)
+     * or specific to a network (<em>twitter_title</em>, <em>facebook_description</em>, etc).
+     */
+    values: {
+      type: Object,
+      default: () => ({}),
```
Comment (caro3801) [inline lib/components/SharingOptions.vue:114]:
```suggestion
      default: () => {},
```

File: lib/components/ScaleLegend.vue [hunk 4]
```diff
@@ -50,12 +50,15 @@ export default {
...
     return {
-      cursorWrapperOffset: 0
+      cursorWrapperOffset: 0,
+      mounted: false
     }
   },
   async mounted () {
     await this.$nextTick()
     this.setCursorWrapperOffset()
+    this.setColorScaleCanvas()
```
Comment (caro3801) [inline lib/components/ScaleLegend.vue:60]: The return statement is not used. It might be because of the refactor, moving the method from "computed" to "methods"

File: lib/components/EmbeddableFooter.vue [hunk 5]
```diff
@@ -12,173 +16,183 @@
...
+    homeUrl: {
+      type: String,
+      default: () => config.get("app.home"),
+    },
+    /**
+     * Sharing option values to bind to the sharing-options component on the bottom-right corner.
+     */
+    sharingOptionsValues: {
+      type: Object,
+      default: () => ({}),
```
Comment (caro3801) [inline lib/components/EmbeddableFooter.vue:102]:
```suggestion
      default: () => {},
```

**Review by caro3801 — CHANGES_REQUESTED** [review]

File: lib/components/EmbeddableFooter.vue (hunk 5 above)
Comment (pirhoo) [reply lib/components/EmbeddableFooter.vue:102]: It's not the same: 

* `default: () => ({})` returns an empty object
* `default: () => {}` returns null

File: lib/components/ScaleLegend.vue (hunk 4 above)
Comment (pirhoo) [reply lib/components/ScaleLegend.vue:60]: The return statement is not needed indeed. Removing it.

Comment (pirhoo) [reply lib/components/ScaleLegend.vue:60]: https://github.com/ICIJ/murmur/pull/62/commits/c36bd8ea69fed381c11dc5932d4cc57c95013226

**Review by caro3801 — APPROVED** [review]

File: lib/components/DonateForm.vue [hunk 6]
```diff
@@ -1,58 +1,113 @@
...
+            class="col donate-form__payment__level"
+            :class="{ active: level == 'rules' }"
+            @click="selectLevel('rules')"
+          >
+            <h3 class="donate-form__payment__heading text-uppercase font-weight-bold text-primary h5">
+              {{ $t('donate-form.benefits.impacts.rules.heading') }}
+            </h3>
+            <div class="Article">
+              <div>  
+                <!-- eslint-disable vue/no-v-html -->
```
Comment (caro3801) [inline lib/components/DonateForm.vue:56]: Maybe adding this rule could help: 'vue/no-v-html': 'off'

Comment (pirhoo) [reply lib/components/DonateForm.vue:56]: In fact it was useful for the SignupForm component (output printed out from an HTTP request to Mailchimp) so I rather keep it.

Comment (caro3801) [reply lib/components/DonateForm.vue:56]: Ok :+1: nice job!
